### DISCUSSION TITLE
Can't snooze "predicted low alerts" sometimes

### DISCUSSION THREAD

**Discussion by xitation** [NightscoutFoundation/xDrip#3673] [Bug reports]
Hey team,

This has been an issue forever but I've finally decided to try and fix it.

Sometimes, and I can't work out when/why, when an alert for a predicted low triggers, it will present the alert and play the alert sound etc, I'll snooze the alert which appears to work correctly, the alert will be gone from the Android notification bar etc, but the alert audio will continue to play.

Other times snoozing the alert works as expected audio stops, snoozes alert for 15 minutes, notifications go away.

I've found no way to fix this playing alert audio after this happens, it just loops forever until a device reboot. 

Stopping the app doesn't stop the alert audio, a hard reboot is the only fix. 

This appears to only happen on the predicted low alerts, and I'm unsure if it's a coincidence but it does tend to happen more often while I'm listening to music or a podcast (might be un-related).

Keen to understand what you need from me to help troubleshoot and bottom out the cause of this issue.

**Comment by Navid200** [comment]
What is the phone brand and model?

**Reply by xitation** [reply]
Google Pixel 8 Pro

**Comment by Navid200** [comment]
This is related: https://github.com/NightscoutFoundation/xDrip/discussions/1921

**Reply by xitation** [reply]
Ah cool, didn't see that one, happy to work on that thread if you prefer?

**Reply by Navid200** [reply]
What's good is that I also have a Pixel.  But, mine is 6a and I am running Android 15 beta.

I am open to suggestions.
It may help if we ask everyone who has experienced this to provide details.  For example, what other apps they run on their phone.  Exactly what model and brand they are running.  What Android version they are using.

We can then put all of that in a table.  Sometimes, this may help us see what is common.

I have been using xDrip since 2018 on different phones and I have never  experienced this.
We have had problems in the past that we found out were caused by a particular app installed on the phone.

**Reply by xitation** [reply]
I've had this issue on all my devices for ages, was just lazy and never raised it until now. 

Happy to fetch whatever info you think will help. 

Also I'll try and grab logs next time it happens.

**Reply by xitation** [reply]
Noting I've been running xDrip since DexDrip :) and moved to xDrip+ a long time ago.

**Reply by xitation** [reply]
I suspect also, because killing/force stopping xDrip doesn't stop the notification it might be the Android notification framework is somehow confused based on how xDrip is triggering the notification. As I'd assume if it's xDrip doing it, killing it entirely should kill the alert sound.

**Reply by xitation** [reply]
Also just for reference it appears that it can happen even when not using Bluetooth audio, so I don't think it's the Bluetooth stuff causing it. Though I do think it might be related to other audio events or call events, as it appears to tend to happen if you are on a call, getting a call or if you a listening to music/podcasts etc so there's some kind of audio event ongoing or commencing around the time it happens.

**Reply by Navid200** [reply]
This has happened to you on multiple phones.  This has never happened to me on multiple (6-7) phones.
What is different?  You and I.  You use a set of apps.  I use a different set of apps.

Can you think of an app that you always install on all your phones?

**Reply by xitation** [reply]
Yeh had it happen on last phone too (Pixel 5). All google devices. In case that is relevant :)

**Reply by xitation** [reply]
> This has happened to you on multiple phones. This has never happened to me on multiple (6-7) phones. What is different? You and I. You use a set of apps. I use a different set of apps.
> 
> Can you think of an app that you always install on all your phones?

haha where to start, I have heaps of apps on there. 

it does appear to happen most often while listening to audio on Bluetooth headphones, I previously used Google Podcasts a lot, though since that was deprecated I've moved to Pocket Casts. I use Youtube Music a bunch too.

**Reply by Navid200** [reply]
I don't listen to music on my phone.  
I wonder if you can switch and use a different app for listening to music and see if you can identify which app may cause it and which alternative app is safe.

**Reply by xitation** [reply]
I can try, though noting it's happened with different apps over the years I suspect it might not be a specific app thing. 
Google Music, became YouTube Music.
I used to use DogCatcher for podcasts, I don't use that anymore
I used to use Google Podcast, now switched over to Pocket Casts. 

Through all these changes it still sometimes happens. 

I mean not ruling out all the other apps with background services that are completely un-related to audio that are present.
